### ISSUE TITLE
[ACME] Ability to throttle new certificate requests

### ISSUE THREAD

**Issue by kallisti5** [containous/traefik#3653]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

It would be awesome to automatically (or manually via a configuration option) throttle ACME certificate requests.   LetsEncrypt enforces several rate limits which when hit can severely impact your ability to request certificates.

Even a basic overall "delay per certificate request" would help keep from crossing rate limits. (As an example, 40 seconds between certificate requests would allow continuous certificate generations without hitting limits)

**Comment by Southclaws** [comment]
This is a must, I just added 8 new subdomains to my infrastructure and now Traefik is repeatedly printing out `Error creating new order :: too many failed authorizations recently: see https://letsencrypt.org/docs/rate-limits/"`

Literally all my sites are offline now - do I wait? do I shut down Traefik and just take everything offline? How long do I have to wait? Seems like a pretty big flaw that it can cause major outage like this.

**Comment by mukesh-urbanpiper** [comment]
@Southclaws  I have the same use-case. Just wondering whether you found any workaround so that it doesn't hit the rate limit.

**Comment by Southclaws** [comment]
I found no workaround, I had to take every site off HTTPS for 1 week which broke a lot of APIs and annoyed a lot of people.

**Comment by derjohn** [comment]
ATM Let's encrypt throttle to 300 requests in 3 hours. You should be able to request 8 at once. You triggered the rate limit by other means, e.g. a different LE client on the same machine (certbot or so ?). You should consider closing that issue, it's old.

**Comment by Southclaws** [comment]
No, the only thing running was Traefik, I wasn't and don't use any other LE clients.

**Comment by ssipos90** [comment]
If for some reason the DNS fails or it's improperly configured, then traefik will keep hitting letsencrypt like crazy until rate limited and then continue to do so. A configurable back-off strategy would be great.
